### PR TITLE
Switch to newer avail-core.

### PR DESCRIPTION
**Pull request by luka-ethernal**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1426,8 +1426,8 @@ dependencies = [
 
 [[package]]
 name = "da-primitives"
-version = "0.2.2"
-source = "git+https://github.com/maticnetwork/avail-core.git?tag=v0.3.0#0025bb53b4190d5920ad351253ed75e078c3c833"
+version = "0.4.0"
```
Comment (MiguelDD1) [inline Cargo.lock:1429]: Should not `da-primitives` be also `0.5.0` following the tag?

**Review by MiguelDD1 — CHANGES_REQUESTED** [review]
Please merge into `develop` branch.

File: pallets/system/src/limits.rs [hunk 2]
```diff
@@ -155,8 +155,8 @@ impl BlockLength {
 		const_assert!(is_chunk_size_valid(BLOCK_CHUNK_SIZE));
 		Self {
 			max: PerDispatchClass::new(|_| max),
-			cols: MAX_BLOCK_COLUMNS,
-			rows: MAX_BLOCK_COLUMNS,
+			cols: MAX_BLOCK_COLUMNS.0,
+			rows: MAX_BLOCK_ROWS.0,
```
Comment (aterentic-ethernal) [inline pallets/system/src/limits.rs:159]: This was a bug?

Comment (luka-ethernal) [reply pallets/system/src/limits.rs:159]: I guess so. I changed it along the way.

**Pull request closed without merging**